app/internal_packages/category-mapper/lib/preferences-category-mapper.tsx

```diff
@@ -72,6 +84,19 @@ export default class PreferencesCategoryMapper extends React.Component<
     );
   };
 
+  _setStateContainerFolders = (container) => {
+    const containerFolders = Object.assign(JSON.parse(JSON.stringify(this.state.containerFolders)), container);
+    this.setState({ containerFolders });
+  }
+
+  _updateContainerFolderDefault = () => {
+    Actions.updateContainerFolderDefault(this.state.containerFolderDefault);
+  }
+
+  _updateAccount = (accountId, updates: Partial<Account>) => {
+    Actions.updateAccount(accountId, updates);
+  }
+
```

Comment:
Overall this looks great - I think this is a tricky setting though because for accounts that require a non-default value (Fastmail for now, maybe others too?) you should ideally set it before starting account sync. Changing it after using the account for a little while could also have strange effects, since you could end up with both Mailspring/Snoozed and CustomPrefix/Snoozed, and right now the sync engine actually "discovers" the default one first unless you go and delete it after creating the custom one.... It would also remove all your currently snoozed messages I think, which might be a little whacky.

What if we moved this setting to the "Connect an Account" screen where you enter the IMAP/SMTP credentials? I think having a "Custom Folder Prefix" option there might be a good way to surface this. (We could try to find other providers later where it needs to be set and update the per-domain template JSON that feeds that screen with defaults).

Let me know what you think!